devicetypes/vision-raytseng/vision-4-in-1-motion-sensor.src/vision-4-in-1-motion-sensor.groovy

```diff
@@ -0,0 +1,510 @@
+		if (state."${getWakeUpInfo("name")}" == value) {
+			if (state."${getWakeUpInfo("name")}Refresh" == true) {		
+				state."${getWakeUpInfo("name")}Refresh" = false
+			}
+		}
+	}
+}
+
+def getWakeUpInfo(text) {
+	def wakeUp = [:]
+	
+	wakeUp.name = "wakeUpInterval"
+	wakeUp.title = "Wake up interval [Hours]"
+	wakeUp.description = ""
+	wakeUp.type = "number"
+	wakeUp.range = "1..4660"
+	wakeUp.default = 24
+	
+	return wakeUp."${text}"
+}
+
+private getCommandClassVersions() {
+	[
+		0x5E: 2,
+		0x22: 1,
+		0x85: 2,
+		0x59: 3,
+		0x80: 1,
+		0x70: 2,
+		0x5A: 1,
+		0x7A: 5,
+		0x87: 3,
+		0x72: 2,
+		0x8E: 3,
+		0X31: 5,
+		0x71: 3,
+		0x73: 1,
+		0x98: 1,
+		0x9F: 1,
+		0x6C: 1,
+		0x55: 2,
+		0x86: 3,
+		0x84: 2 
+	]
+}
+
+def parse(String description) {
+	def result = []
+	def cmd = zwave.parse(description, commandClassVersions)
+	
+	if (cmd) {
+		result += zwaveEvent(cmd)
+	}
+	else {
+		logDebug "Unable to parse description: ${description}"
+	}
+	
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	def cmds = []
+	
+	cmds += configure()
+	if (cmds) {
+		cmds << "delay 5000"
+	}
+	cmds << zwave.wakeUpV2.wakeUpNoMoreInformation().format()
+	
+	return cmds ? response(cmds) : []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	if (cmd.nodeid == zwaveHubNodeId) {
+		def interval = [:]
+		interval."${getWakeUpInfo("name")}" = second2Hour(cmd.seconds)
+		wakeUpIntervalCheck(interval)
+	}
+	[]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def map = [name: "battery", unit: "%"]
+	
+	if (cmd.batteryLevel == 0xFF) {
+		map.value = 1
+		map.descriptionText = "${device.displayName} has a low battery"
+		map.isStateChange = true
+	} else {
+		map.value = cmd.batteryLevel
+	}
+	
+	state.lastBatteryReport = new Date().time
+	createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def size = getConfigurationInfo(cmd.parameterNumber, "size")
+	def value = array2Integer(cmd.configurationValue)
+	def config = [:]
+	
+	if (size && size == cmd.size) {
+		switch (getConfigurationInfo(cmd.parameterNumber, "type")) {
+			case "enum":
+				def optionName = getConfigurationInfo(cmd.parameterNumber, "enumMap").find { it.value == value}?.key
+				if (optionName) {
+					config."${getConfigurationInfo(cmd.parameterNumber, "name")}" = optionName
+				}
+				break
+				
+			case "number":
+				config."${getConfigurationInfo(cmd.parameterNumber, "name")}" = value
+				break
+		}
+		
+		if (config) {
+			configurationCheck(config)
+		}
+	}
+	[]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	def result = []
+	
+	if (cmd.notificationType == 0x07) {
+		if (cmd.eventParametersLength) {
+			cmd.eventParameter.each {
+				if (it == 0x03) {
+					result = createEvent(name: "tamper", value: "clear") 
+				}
+				else if( it == 0x08) {
```

Comment:
Please use this convention of formatting (applies to all occurrences)
`} else if() {
...
}`